### PR TITLE
Closes #2272: Adds removeProvider and removeAllProviders to AwesomeBar

### LINKED ISSUE #2272: Allow SuggestionProviders to be removed from BrowserAwesomeBar

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by sblatz** [comment]
In hindsight, I think we need to also remove the existing suggestions from the recyclerView.

File: components/browser/awesomebar/src/main/java/mozilla/components/browser/awesomebar/SuggestionsAdapter.kt [hunk 1]
```diff
@@ -65,6 +65,13 @@ internal class SuggestionsAdapter(
         updateTo(updatedSuggestions.sortedByDescending { it.score })
     }
 
+    fun removeSuggestions(provider: AwesomeBar.SuggestionProvider) = synchronized(suggestions) {
+        // Start with the current list of suggestions
+        val updatedSuggestions = suggestions.toMutableList()
+        suggestionMap[provider]?.let { updatedSuggestions.removeAll(it) }
+        updateTo(updatedSuggestions)
+    }
```
Comment (pocmo) [inline components/browser/awesomebar/src/main/java/mozilla/components/browser/awesomebar/SuggestionsAdapter.kt]: I wonder if we could add a test for this method to `SuggestionsAdapterTest`?

File: components/browser/awesomebar/src/test/java/mozilla/components/browser/awesomebar/BrowserAwesomeBarTest.kt [hunk 2]
```diff
@@ -133,6 +134,43 @@ class BrowserAwesomeBarTest {
...
+            awesomeBar.addProviders(provider1, provider2)
+            awesomeBar.removeAllProviders()
+
+            awesomeBar.onInputStarted()
+
+            verify(provider1, never()).onInputStarted()
+            verify(provider2, never()).onInputStarted()
+        }
+    }
+
```
Comment (pocmo) [inline components/browser/awesomebar/src/test/java/mozilla/components/browser/awesomebar/BrowserAwesomeBarTest.kt]: Those test cases are great. Maybe we could also test that removing a provider will call `removeSuggestions` on the adapter? In this file you'll find examples where we replace the adapter with a mock to verify calls on the adapter.

Comment (sblatz) [reply components/browser/awesomebar/src/test/java/mozilla/components/browser/awesomebar/BrowserAwesomeBarTest.kt]: Good ideas on these tests--I added them!

**Pull request merged**